students/Dkuchan/session6/classessheet.py

```diff
@@ -0,0 +1,51 @@
+# classessheet.py
+""" This is a script to mess with classes.
+    The idea is to use this practice to create class "persons" for mail room """
+
+
+class Donor:
+    # This is an experimental class for donors in mailroom
+    numdonors = 0
+    rosterraw = []
+
+    def __init__(self, name):
+
+        self.first_name, self.last_name = list(name.split())
+        self.donations = [1, 2, 3, 4, 5, 6, 7]
+        self.fullname = str(self.first_name + ' ' + self.last_name)
+        self.averagedon = sum(self.donations) / len(self.donations)
+        self.maxdon = max(self.donations)
+        Donor.numdonors += 1
+        Donor.rosterraw.append(id(self))
+
+    def updatecalcs(self):
+        # Updates the average donation attribute after mods have been made to donations
+        self.averagedon = sum(self.donations) / len(self.donations)
+        self.maxdon = max(self.donations)
+
+    def lastfirst(self):
+        # creates a last name , first name string
+        lastfirstname = '{}, {}'.format(self.last_name, self.first_name)
+        return lastfirstname
+
+    def appenddonations(self, newdon):
+        ''' Adds a donation to the donations list
+            Updates the average donations
+            Updates the max donation '''
+        self.donations.append(newdon)
+        self.updatecalcs()
+
```

Comment:
I think your Donor class would be easer if each instance represented a single donor, and then these donors were put into a database or another class. There is quite a bit of awkwardness here that I think could be sorted out by separating individual donor info from the group. The code seems to be broken, btw, because you increase numdonors when you initialize, which means Dan is in position 1 of your donor list (with None still in position 0), but when you add Abbey, she is 2 and the donor list you created only has 2 positions, so she cannot be added to your list.